backend/src/apiserver/server/experiment_server_test.go

```diff
@@ -422,3 +428,50 @@ func TestValidateCreateExperimentRequest_Multiuser(t *testing.T) {
 		}
 	}
 }
+
+func TestArchiveAndUnarchiveExperiment(t *testing.T) {
+	// Create experiment and runs under it.
+	clients, manager, experiment := initWithExperimentAndPipelineVersion(t)
+	defer clients.Close()
+	runServer := NewRunServer(manager)
+	run1 := &api.Run{
+		Name:               "123",
+		ResourceReferences: validReferencesOfExperimentAndPipelineVersion,
+	}
+	err := runServer.validateCreateRunRequest(&api.CreateRunRequest{Run: run1})
+	assert.Nil(t, err)
+	_, err = runServer.CreateRun(nil, &api.CreateRunRequest{Run: run1})
+	assert.Nil(t, err)
+	clients.UpdateUUID(util.NewFakeUUIDGeneratorOrFatal(resource.FakeUUIDOne, nil))
```

Comment:
`clients.UpdateUUID(util.NewFakeUUIDGeneratorOrFatal(resource.FakeUUIDOne, nil))`

For my own learning, this is necessary to allow creating two runs, right?

Reply:
Yes. this updateUUID is a (temporary) hack because we want the two runs being assigned two different UUID. Our current fake uuid generator keeps emitting the same uuid (or the uuids in an array one by one). So if we are going to create a second run. We insert a different uuid in and the second run created afterwards will have a different uuid.

There can be some improvement to our fake uuid generator, e.g., to make it emit random fixed length ids instead of emitting the same uuid again and again. This way, we don't need to update uuid in the middle or manually prepare an array of uuids for it to emit. If we found cycles in our dev team, we can prioritize this work item to help with our productivity.

Reply:
Thanks for the detailed explanation!